### PR TITLE
Create site split for iowasummerwritingfestival.uiowa.edu.

### LINKED ISSUE #5886: ISWF event customizations
- [x] Add "presenters" list to subEvents blocks between date and short description - icon
- [x] Move "presenters" above description in event display.
- [x] Change short dates to long dates to match standard moving forward: l, F j, Y - g:ia
- [x] event type - Select list custom to ISWF - allowed values (no longer necessary??, as per https://github.com/uiowa/uiowa/issues/5732#issuecomment-1263915168, unless we need this for conditional display of fields later)
- [x] Add Genre - only for workshops??
- [x] Display genre(s) below descriptions in a comma-separated list.
- [x] Representing Performers - label as presenter for now - future slice to make this customizable
- [x] Add registration link and add as button below details in right column

### LINKED ISSUE #5931: ISWF View Listing
# Issue

Create searchable view listing for events by date, genre, and presenter.

# Proposed Solution

- [x] Try to use existing view display otherwise create a ISWF specific view, adding it to the site split.
  - Just filtered events that are not referenced as a parent. So, if there are multiple types this applies to, the view might need to look at certain relationships ("lowest layer")
- [x] As a visitor, I want to pick a date or date range (depending on what we discover while building) that I am available and see events that meet that criteria (on that date or in-progress).
- [x] As a visitor, I want to filter by genre
- [x] Filtered dropdown or autocomplete of person published (not hide on lists) that have been referenced by an event. This would require an entity reference view display to filter so that not all person content on the website is available for selection.
- [x] Ensure consistency of different event listing displays. Excludes event list block because that has an explicit option to change how it looks so that is handled differently.
- [x] If the "Performer" person type is not necessary to create the above views, remove it from the ISWF site split.
- [x] Based on today's conversation (12/05/2022), add an image field for visual conformity. To do this, you’ll want to use the default site and then turn on the event feature.
- [x] Look into genre dropdown opening when "Apply" is clicked.
- [x] Date filter should show events that have date ranges including the date selected.

### PR DESCRIPTION
**Pull request by mark-bennett-uiowa**
Creation of site split for ISWF.

Resolves #5931
Resolves #5886 

```
ddev blt ds --site=iowasummerwritingfestival.uiowa.edu && ddev drush @iowasummerwritingfestival.local config-split:activate event -y && ddev drush @iowasummerwritingfestival.local en better_exposed_filters -y && ddev drush @iowasummerwritingfestival.local config:import --source ../config/sites/iowasummerwritingfestival.uiowa.edu --partial -y && ddev drush @iowasummerwritingfestival.local cim -y
```

### Testing ISWF
- SETUP  
    - `ddev composer install && ddev blt frontend`
    - `ddev blt ds --site=iowasummerwritingfestival.uiowa.edu`
    - `ddev drush @iowasummerwritingfestival.local config-split:activate event -y`
    - `ddev drush @iowasummerwritingfestival.local en better_exposed_filters`
    - `ddev drush @iowasummerwritingfestival.local config:import --source ../config/sites/iowasummerwritingfestival.uiowa.edu --partial -y && ddev drush @iowasummerwritingfestival.local cim -y`
- TESTING
   - Create a new local event and populate with all relevant fields (this includes featured image, short description in sidebar)
      - The genre field should display in two column format. 
   - Ensure that our ISWF custom fields are rendering correctly (event type, genre, and registration link).
   - Ensure that the date display is using the smartdate Default - Long date format.
   - Ensure that the genre and event type fields contain various options and are not empty.
   - Investigate Console display of code, and search for ld+json. Ensure that registration link is mapped to schema.org via the the offers property (will appear in one of the two ld+json sections.
   - Create additional events that have varying empty relevant fields (event type, genre, registration link, short description) and reference original event to test the sub event addition. These new "sub" events should show in "Schedule" region of the first created event's display.
      - The subevent teasers under Schedule should show Presenters comma-separated list.
   - Add or edit a person entity via Content, ensure that 'Presenter' is an option under Person Types.
   - Sticky events, if part of the returned results, should show atop the events_list_block events and sub-events views.

### Testing Default
- SETUP  
    - `ddev composer install && ddev blt frontend`
    - `ddev blt ds`
    - `ddev drush @default.local config-split:activate event -y && ddev drush @default.local cim -y`
- TESTING
    - Follow a similar strategy to the Testing list above for ISWF and ensure this does not break default implementation.

Initial work:

- [x] Moved presenters above description in default event layout.
- [x] Added **field_event_type** single value field. Populated with 'Workshop' to avoid errors preventing future work.
- [x] Added **field_event_genre** field unlimited values field. Populated with previous genres from existing site to avoid errors and test display.
- [x] Added **field_registration_link** field for internal/external links, no title.
- [x] Styled **field_registration_link** field as a button in second column.
- [x] Created 'Default long date' smart date format.
- [x] Changed dates on events view and display layout to 'Default long date' version over 'default'.
- [x] populate field_event_type with allowed values function from custom site module.
- [x] populate field_event_genre with allowed values function from custom site module.
- [x] add field_registration_link field value to offers property in schema.org structure.
- [x] field_registration_link button needs moved above bordered block or fix styling to fit inside current region.
- [x] added 'presenter' person type with performer machine_name to match schema in case we change the label at a later time. (added in single push so can remove if later determined)

ToDo (not complete list):

- [x] Change field_event_genre form display to two column list like Person Type (may not be necessary).
- [x] ~field_event_genre list will not show as comma-delimited in view display, changed back to list for now.~ Removed per client feedback.
- [x] ~Remove field_event_type from view display~ (not necessary)
- [x] Set order of sidebar "meta" information to Location, Virtual Link, Source Link (combination of sitenow_event feature split and this site split)
- [x] Add spacing between meta field displays
- [x] Show event image as widescreen rather than square
- [x]  Update sidebar to use new card template with view that generates a card so that we can separate out styles and code. remove old styling.
- [x] Do not show event image if one is not available, change default not shown away from empty person image
- [x] Update genre list based on customer feedback. Verified with client on 11-14.
- [x] Update event type list based on customer feedback. Verified with client on 11-14.
- [x] Add presenters to sub event view blocks between date and description

### Lazy setup commands
<details>
<summary>with ddev</summary>

```
ddev composer install && ddev blt ds --site=iowasummerwritingfestival.uiowa.edu && ddev drush @iowasummerwritingfestival.local config-split:activate event -y && ddev drush @iowasummerwritingfestival.local en better_exposed_filters -y && ddev drush @iowasummerwritingfestival.local config:import --source ../config/sites/iowasummerwritingfestival.uiowa.edu --partial -y && ddev drush @iowasummerwritingfestival.local cim -y && ddev blt frontend && ddev drush @iowasummerwritingfestival.local cr
```

</details>

<details>
<summary>without ddev</summary>

```
composer install && blt ds --site=iowasummerwritingfestival.uiowa.edu && drush @iowasummerwritingfestival.local config-split:activate event -y && drush @iowasummerwritingfestival.local en better_exposed_filters -y && drush @iowasummerwritingfestival.local config:import --source ../config/sites/iowasummerwritingfestival.uiowa.edu --partial -y && drush @iowasummerwritingfestival.local cim -y && blt frontend && drush @iowasummerwritingfestival.local cr
```

</details>

### CODE REVIEW

**Comment by joewhitsitt** [comment]
```
FILE: ...summerwritingfestival_core/iowasummerwritingfestival_core.module
----------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 4 LINES
----------------------------------------------------------------------
 26 | ERROR | [x] TRUE, FALSE and NULL must be uppercase; expected
    |       |     "NULL" but found "null"
 27 | ERROR | [x] Opening brace should be on the same line as the
    |       |     declaration
 79 | ERROR | [x] TRUE, FALSE and NULL must be uppercase; expected
    |       |     "NULL" but found "null"
 80 | ERROR | [x] Opening brace should be on the same line as the
    |       |     declaration
```

https://github.com/uiowa/uiowa/pull/5925/files#diff-0399c32055fda1c98719b855184e2629d61c80d1fb92368bbcb4a778c6ea162dR23 - I think this should be a partial item for the site split.

I believe only the field.storage items need to be declared and that the field instances are automatically included.

Can the option keys be free of spaces? (e.g. Social Event becomes SocialEvent) This removes any potential issues with encoding and calling these in code.

File: docroot/themes/custom/uids_base/templates/field/field--node--field-event-registration-link--event--default.html.twig [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * - field_type: The type of the field.
+ * - label_display: The display settings for the label.
+ *
+ *
+ * @see template_preprocess_field()
+ */
+#}
+
+{% for item in items %}
+  <a href="{{ item.content['#url'] }}" title="{{ item.content['#title'] }}" role="button" class="bttn bttn--full bttn--primary">
```
Comment (joewhitsitt) [inline docroot/themes/custom/uids_base/templates/field/field--node--field-event-registration-link--event--default.html.twig]: Why add the title attribute but then hardcode the text?

`<a href="{{ item.content['#url'] }}" class="bttn bttn--full bttn--primary">{{ item.content['#title']|render }}</a>` - for reference docroot/sites/grad.admissions.uiowa.edu/modules/grad_admissions_core/templates/field--node--field-area-of-study-apply-link--area-of-study.html.twig

**Comment by joewhitsitt** [comment]
>I believe only the field.storage items need to be declared and that the field instances are automatically included.

>welp. maybe i am not correct. you can leave it for now or experiment if you want

**Comment by mark-bennett-uiowa** [comment]
> ...ds_base/templates/field/field--node--field-event-registration-link--event--default.html.twig

It seems that iisc and immuno.grad do this and seem to be successful, so Im giving it a shot. Can revert if needed.

**Comment by mark-bennett-uiowa** [comment]
> I believe only the field.storage items need to be declared and that the field instances are automatically included.
 
It seems that iisc and immuno.grad do this and seem to be successful, so Im giving it a shot. Can revert if needed.

File: docroot/themes/custom/uids_base/templates/field/field--node--field-event-registration-link--event--default.html.twig (hunk 1 above)
Comment (mark-bennett-uiowa) [reply docroot/themes/custom/uids_base/templates/field/field--node--field-event-registration-link--event--default.html.twig]: I changed the title parameter for the <a> tag to match the hardcoded value. The title was empty anyways, which would have left the parameter value as empty, but good catch!

Comment (joewhitsitt) [reply docroot/themes/custom/uids_base/templates/field/field--node--field-event-registration-link--event--default.html.twig]: is the field collecting a title or is it link only?

Comment (mark-bennett-uiowa) [reply docroot/themes/custom/uids_base/templates/field/field--node--field-event-registration-link--event--default.html.twig]: link only, title is hidden from the form. Didn't want the user to get "creative" with the what the link button title.

Comment (joewhitsitt) [reply docroot/themes/custom/uids_base/templates/field/field--node--field-event-registration-link--event--default.html.twig]: Okay

**Comment by joewhitsitt** [comment]
We weren't sure where Alan's local was in terms of the todos so we pressed forward. It looks like there are still some discussion needed about the remaining checkboxes here:

> Change field_event_genre form display to two column list like Person Type.
>Update genre list based on customer feedback.
>Add presenters to sub event view blocks between date and description (may not be necessary, checking with client)

File: docroot/themes/custom/uids_base/scss/components/card.scss [hunk 2]
```diff
@@ -306,6 +306,12 @@
   margin-top: .3rem;
 }
 
+.card__meta {
```
Comment (bspeare) [inline docroot/themes/custom/uids_base/scss/components/card.scss:309]: We could probably move this back to UIDS4 with something like:
```
.card__meta {
  margin-top:.3rem;
  > * {
  margin-top: .3rem;
  }
}
```
I'll make a note in the UIDS4 issue.

**Comment by GaryRidgway** [comment]
Initial code review seems good but there is one change to be made, should be good to demo

**Comment by pyrello** [comment]
There was more refactoring that I wanted to do, but I decided to stop. Hoping to do a bit more consolidation of logic in #5931.

**Comment by bspeare** [comment]
I removed the card border style from the event feature display.  I'd rather see those display as lists to keep our UX for list displays consistent.   The border remains set for this iswf split.

**Comment by bspeare** [comment]
Pulling to test.

**Comment by bspeare** [comment]
Is the "Hide on Lists name" here supposed to be hidden?

Screen Shot 2022-11-18 at 3 45 12 PM

**Comment by bspeare** [comment]
You can reference the same event and it causes some display issues:  Screen Shot 2022-11-18 at 3 46 21 PM

**Comment by mark-bennett-uiowa** [comment]
> Is the "Hide on Lists name" here supposed to be hidden?
> 
> 

Yes, it should be. This could be a by-product of what is done with the person teaser on sitenow_events.module line 330-332.

**Comment by mark-bennett-uiowa** [comment]
> You can reference the same event and it causes some display issues: 
>

Dang, I guess I never tried that. Is it possible to prevent a self-reference through that field?

**Comment by bspeare** [comment]
Is this supposed to hide the event on the sub list?

Screen Shot 2022-11-18 at 3 54 00 PM

**Comment by mark-bennett-uiowa** [comment]
> Is this supposed to hide the event on the sub list?
> 

I dont think that is necessary and could be removed. I assumed that was for Persons only.

**Comment by bspeare** [comment]
Webmasters have access to both so we should either make them work with the views lists that we have or remove them.  We might need more input from the team on this.

Screen Shot 2022-11-18 at 3 57 00 PM

**Comment by bspeare** [comment]
Using the Events List block on oniowa.uiowa.edu it appears we lost the location and virtual details fields from this view.  And we also need to add the ability to hide the "Presenter" within this block.  

Screen Shot 2022-11-18 at 4 09 56 PM

On https://iowasummerwritingfestival.uiowa.ddev.site/, the Events list block does not display events. 

Screen Shot 2022-11-18 at 4 11 13 PM

**Comment by bspeare** [comment]
Testing recap:
- ~"Hide on Lists" when creating a person is not hiding the name on the `events_list_block` or `events` views.~ - This works as expected based on our conversations.
- [ ] Sticky promotion option not working on the `events_list_block` or `events` views.
- [x] `events_list_block` no longer displaying location or virtual details. 
- [x] `events_list_block` does not have an option to hide the presenters field.

**Comment by joewhitsitt** [comment]
Nice testing @bspeare. I think sticky and hide on lists should continue to be supported.

https://www.drupal.org/project/entity_reference_validators is a temp contrib project while https://www.drupal.org/project/drupal/issues/2280479 is being worked on to prevent self-referencing, duplicates.

**Comment by pyrello** [comment]
> https://www.drupal.org/project/entity_reference_validators is a temp contrib project while https://www.drupal.org/project/drupal/issues/2280479 is being worked on to prevent self-referencing, duplicates.

Note to self: Always read @joewhitsitt's responses to save myself from spending time looking up something he already got to faster.

**Comment by pyrello** [comment]
I'm going to pick up on the work that is indicated here that needs to be done.

**Comment by pyrello** [comment]
To clarify: The todos that @bspeare identified apply to the event feature split rather than the ISWF site split. This means that those adjustments need to be made using the default site with the event feature turned on rather than the ISWF split. After those changes are exported, then those changes might need to be absorbed into the ISWF split as well.

**Comment by pyrello** [comment]
> "Hide on Lists" when creating a person is not hiding the name on the events_list_block or events views.

This seemingly simple thing is actually fairly complicated. In other places where we are observing the value of the "Hide on lists" option for listing people, we are listing people in a view and we simply filter out people based on the value of that field. In this case, the person entity isn't actually included in these views and so we don't get to evaluate the field in question. We can potentially redesign this view to use a relationship that would allow us access to the field, although that makes reasoning about what the view should be showing a lot more difficult. I'm wondering if we need to actually filter people with "Hide on lists" checked from being able to be selected when populating the presenters field?

**Comment by pyrello** [comment]
> > Is this supposed to hide the event on the sub list?
> 
> I dont think that is necessary and could be removed. I assumed that was for Persons only.

I think this may just be something that was carried over because of copy/paste error. Just checking in about whether we think there is any value in creating a "hidden" event? I would be inclined to say no at this point because figuring out what that actually means in terms of nested events is going to be a really doozy.

**Comment by joewhitsitt** [comment]
I think i got confused. I don’t consider entity references to to apply here to this functionality. similar to an author person being listed on an article despite having hide on lists checked. so no, I don't think there is extra hide on lists work as it relates to presenters. main goal of the functionality were the general views of this content (view pages, list blocks). though rare, i think this could be an “unlisted event” that is directly linked to but doesn’t show up on the homepage for instance.

The sub events, though a view, are intentionally referenced. so I don't think hide on lists applies.

**Comment by pyrello** [comment]
I'm also noticing that the "Performer" person type is still part of the ISWF site split. I know there was some talk about using it in #5931, but I think we established that using the fact of a person being attached to event as a way of denoting them as a performer would prevent users having to manually set this value and potentially introducing the possibility that those two things are out of sync. I added a bullet point to remove this config in #5931 if we are able to do everything there we need to without it.

**Comment by pyrello** [comment]
> I think i got confused. I don’t consider entity references to to apply here to this functionality. similar to an author person being listed on an article despite having hide on lists checked. so no, I don't think there is extra hide on lists work as it relates to presenters. main goal of the functionality were the general views of this content (view pages, list blocks). though rare, i think this could be an “unlisted event” that is directly linked to but doesn’t show up on the homepage for instance.
> 
> The sub events, though a view, are intentionally referenced. so I don't think hide on lists applies.

Thanks, @joewhitsitt. Wanting to make sure I'm getting what you're saying.
- "Hide on lists" for people does not apply to those people being referenced and therefore getting indirectly listed. So this would be works as expected?
- Are you advocating for trying to figure out the "hidden event" feature now? Or can we safely say that this isn't functionality we want to explore at this point and back out the relevant code that relates to it?

**Comment by joewhitsitt** [comment]
We talked in zoom as well, but for visibility:

>"Hide on lists" for people does not apply to those people being referenced and therefore getting indirectly listed. So this would be works as expected?

Correct. Your idea of help text to make this more clear sounds good to me.

>Are you advocating for trying to figure out the "hidden event" feature now? Or can we safely say that this isn't functionality we want to explore at this point and back out the relevant code that relates to it?

Personally, I think the hide on lists should be added to those views to maintain consistency with our other default content types. Sub-events brings up an interesting situation, but since they are pulled together through another kind of reference (parent event), I don't think it applies to this view.

**Comment by pyrello** [comment]
FWIW, I think that having a "hidden" event that doesn't show on a parent event makes for a poor user experience. I guess I can see excluding it from other lists, but the UX of being able to click on the parent event link from the hidden event and then not seeing that event listed on the parent event doesn't seem right to me.

**Comment by pyrello** [comment]
> Created 'Default long date' smart date format.

Anyone know why there was a special date format created for this split?

**Comment by briand44** [comment]
>Anyone know why there was a special date format created for this split?

I do not but in the issue summary there is a checkbox for 
"Change short dates to long dates to match standard moving forward: l, F j, Y - g:ia"

**Comment by joewhitsitt** [comment]
Not sure if it was @mark-bennett-uiowa's personal preference or if it was a custom request. I do recall Mark talking to @bspeare about before the change was made and that Admissions again was part of the reason the event dates are currently the way they are.

**Comment by GaryRidgway** [comment]
Yeah mark wanted a long date display

**Comment by pyrello** [comment]
Running into this on an event I just created: 
```Error: Call to a member function getOptions() on null in Drupal\smart_date\Plugin\Field\FieldFormatter\SmartDateDefaultFormatter->viewElements() (line 45 of modules/contrib/smart_date/src/SmartDateTrait.php).```

**Comment by pyrello** [comment]
> events_list_block no longer displaying location or virtual details

It seems that this is happened here: https://github.com/uiowa/uiowa/pull/5869/files#diff-af2c7ba26f825b10d8a8016e8191ac2bb8092e13fcf39d415c4996241415ae78L282-L310

**Comment by bspeare** [comment]
> It seems that this is happened here: https://github.com/uiowa/uiowa/pull/5869/files#diff-af2c7ba26f825b10d8a8016e8191ac2bb8092e13fcf39d415c4996241415ae78L282-L310

I think we only meant to remove that for the sub events view.

**Comment by joewhitsitt** [comment]
https://github.com/uiowa/uiowa/pull/5925#issuecomment-1324052881 was similar to the issue @GaryRidgway ran into where the date format was being called in the sidebar view but wasn't available outside of the site split. I thought we resolved that.

**Comment by pyrello** [comment]
> Sticky promotion option not working on the events_list_block or events views.

This is the only item left from this comment: https://github.com/uiowa/uiowa/pull/5925#issuecomment-1320594413. Does a sticky event show before other events listed in chronological order?

The promotion options not working seems to be the result that they were never implemented in the first place. Kudos to @bspeare for finally catching this.

I think I have fixed everything else that was amiss. I also made some adjustments to the order of elements in displays and forms to try to reduce the noise in the configuration patches. 

It is my opinion that the new date format that was created should either be ditched in the ISWF split (unless it is a deal breaker for the client) or it should be adopted as the standard date display for local events throughout the whole feature. If the latter happens, I think this would require adjusting the configuration for Admissions as they were specific about a different date format.

**Comment by joewhitsitt** [comment]
>Does a sticky event show before other events listed in chronological order?

Yes, the view sort would have sticky above the chronological sort, so even if it was a list of upcoming events, a past event marked as sticky would be at the top of the list. If there are more than one sticky events, then chronological sorting takes affect.

**Comment by pyrello** [comment]
> so even if it was a list of upcoming events, a past event marked as sticky would be at the top of the list

I don't think that sticky should override our filter options, which would remove a past event from list blocks currently.

**Comment by joewhitsitt** [comment]
>I don't think that sticky should override our filter options

Ah. yes. Forgot what filters were in place. "of the returned results..."

**Comment by joewhitsitt** [comment]
Adding the sticky to push this forward a bit.

Update: Add in https://github.com/uiowa/uiowa/pull/5925/commits/06117964691f082461b73597f7a655802ce15de8, added a test instruction.

**Comment by mark-bennett-uiowa** [comment]
Pulled for testing.

**Comment by mark-bennett-uiowa** [comment]
I ran some tests to see how the recent changes were working/not-working:

1. The microphone icon for presenters is showing on individual events as well as event lists:
Screen Shot 2022-11-28 at 1 57 09 PM
 
2. The "Sticky at top of lists" bit is active by default when creating a new event. Is this intended?

3. The "Hide on lists" does not seem to have any effect on whether the event shows up or not:
Screen Shot 2022-11-28 at 1 10 19 PM

4. What are we intending to do with the dates? See this screenshot below where the event list block dates are shortened, the individual event meta dates are long, etc.
Screen Shot 2022-11-28 at 1 45 23 PM

If Admissions is the only one wanting a different date format, then shouldn't we add that customization to their split?
Writing new issue for some discussion about date formatting, coming soon...

**Comment by joewhitsitt** [comment]
Looks like the mic needs some padding or margin adjustment.

>The "Sticky at top of lists" bit is active by default when creating a new event. Is this intended?

No it is not. My bad. That can be unchecked on the event content type edit screen. The permissions should stay though to allow editors to override the default.


>The "Hide on lists" does not seem to have any effect on whether the event shows up or not

I think where it ended up (much discussion in this thread, slack, meetings) was that the main events listing and list block were going to honor the hide on lists. The sub events listing (entity referenced inferred) was not going to honor the hide on lists check. https://github.com/uiowa/uiowa/pull/5925/commits/1aef8517d1764d8fb16f7d1c2740609ccab044f1


>What are we intending to do with the dates? 

Sean had an opinion: 

>It is my opinion that the new date format that was created should either be ditched in the ISWF split (unless it is a deal breaker for the client) or it should be adopted as the standard date display for local events throughout the whole feature. If the latter happens, I think this would require adjusting the configuration for Admissions as they were specific about a different date format. - https://github.com/uiowa/uiowa/pull/5925#issuecomment-1324331336

**Comment by bspeare** [comment]
The icon should not be showing up on the full node display, it is only meant for the view list.

**Comment by bspeare** [comment]
https://github.com/uiowa/uiowa/pull/5925/commits/77c68cdc8b57d918d8d233228bce9f883341c5f2 should fix the issue noted about the icon in https://github.com/uiowa/uiowa/pull/5925#issuecomment-1329680019.

**Comment by pyrello** [comment]
I'm going to test this related to #5931.

**Comment by pyrello** [comment]
The placeholder labels for the start and end dates were swapped, so I fixed that. I also removed BEF as a dependency of ISWF's site split, since it is already covered by the event feature split and the only usage it has in ISWF is where it overrides config from that feature. This allowed partially splitting the events view, which I then cleaned up to remove unintentional differences from the base config.

I was also not able to get the div for the image to not display when there was no image using the existing card template. To avoid affecting all the other card implementations and ballooning our testing requirements for this, I implemented a temporary workaround. We can work on resolving the differences in #5971 or #5970.

**Comment by pyrello** [comment]
> The placeholder labels for the start and end dates were swapped, so I fixed that.

@GaryRidgway explained the rationale for why the labels don't match what is being checked. I changed it back, tested it out and it worked as expected. So this was not a mistake and my apologies for indicating as much. It is actually a very clever solution for handling various event range conditions.

**Comment by bspeare** [comment]
Screen Shot 2022-12-08 at 1 27 07 PM

Do we want this to close when we click outside the toggle?

**Comment by bspeare** [comment]
Screen Shot 2022-12-08 at 1 28 59 PM

Should we bold this "Select None" when it is selected/focused?  Maybe add an underline for at least the focus state?

**Comment by pyrello** [comment]
Going to check all these changes against Admissions.

**Comment by pyrello** [comment]
It looks like we have an Admissions styling regression:
Screen Shot 2022-12-08 at 1 58 54 PM

Here's what it looks like in production:
Screen Shot 2022-12-08 at 1 59 03 PM

Note: I did run `blt frontend`.

**Comment by pyrello** [comment]
> It looks like we have an Admissions styling regression

For the missing border, this is because we switched the main event type display to use the events detail view for the sidebar and removed the underlying style rule that was applying the border directly to the region in the layout template.

Do we want to re-apply the style rule and scope it to Admissions? Or do we want to update Admissions to use the events detail view?

**Comment by GaryRidgway** [comment]
> > It looks like we have an Admissions styling regression
> 
> For the missing border, this is because we switched the main event type display to use the events detail view for the sidebar and removed the underlying style rule that was applying the border directly to the region in the layout template.
> 
> Do we want to re-apply the style rule and scope it to Admissions? Or do we want to update Admissions to use the events detail view?

I think we should update admissions to use the detail view. Should reduce maintenance in the future

Edit: Sean seems to have accomplished this. Sorry for brain lag

**Comment by GaryRidgway** [comment]
> Screen Shot 2022-12-08 at 1 27 07 PM
> 
> Do we want this to close when we click outside the toggle?

I would love this, but I am unsure of how to do so? is there a n option in views that sets this?

**Comment by bspeare** [comment]
@GaryRidgway good question, I wasn't sure if we had control over that.

**Comment by GaryRidgway** [comment]
> @GaryRidgway good question, I wasn't sure if we had control over that.

I just looked through views and couldn't find one. The only thing i can think of is adding a JS detector that checks when you unfocus and simulates a click on the filter? perhaps that is a later enhancement? The bolding is a good quick user enhancement though, I can do that

**Comment by bspeare** [comment]
@GaryRidgway it might be interesting to see if there's any issues on drupal.org that deal with that issue.

**Comment by GaryRidgway** [comment]
> @GaryRidgway it might be interesting to see if there's any issues on drupal.org that deal with that issue.

Just searched through them, not seeing any right now, but i may have missed something

**Comment by pyrello** [comment]
> > @GaryRidgway it might be interesting to see if there's any issues on drupal.org that deal with that issue.
> 
> Just searched through them, not seeing any right now, but i may have missed something

Might be a good feature request for BEF that we could collaborate on during a profdev pair coding session!

**Review by pyrello — APPROVED** [review]
Tested event features, ISWF split, Admissions, and IISC. Looks good.

**Pull request merged**